Adding tWZj_NLO_5f cards to production/13p6TeV

Wilsker opened the pull request:
Adding the cards for the tWZj NLO 5F process. Currently no-b-mass model is default. Have also sent model with restriction cards to conveners. Model and cards have been scrutinised and tested by @agrohsje, @sihyunjeon and @Wilsker.

menglu21 commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_proc_card.dat:
```diff
@@ -0,0 +1,42 @@
+# Re-define proton and jet to remove b-quark contribution from proton/jet
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
```
these four lines may be not needed

menglu21 replied:
yes, sorry, again i didn't point them out clearly, i meant line 36 to line 39

menglu21 commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_run_card.dat:
```diff
@@ -0,0 +1,285 @@
+#*********************************************************************
+# Minimum and maximum pt's (for max, -1 means no cut)                *
+#*********************************************************************
+  20.0	= ptj ! minimum pt for the jets
```
won't it be too tight for a LHE cut on jet, i would prefer 10 GeV

Wilsker replied:
The targeted topology will require the forward jet in tWZj to have a pt > 30 GeV

Wilsker commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_proc_card.dat:
```diff
@@ -0,0 +1,42 @@
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+# Model restricted to massless b-quarks
+import model Yt_UFO-nobmass
```
This is the model that will be loaded, the '-nobmass' should ensure the restriction card 'restrict_nobmass.dat' is implemented, which I have provided @agrohsje and @sihyunjeon. I believe they will upload the updated model directory soon

sihyunjeon replied:
You have to add "extramodels.dat"

I replaced UFO model with restriction cards in the web. Yt_UFO.tar.gz
FYI I set everything that's lighter than bottom to 0 as it doesn't make sense to treat u quark for example to be massive while mb=0. Also lightly tested in my MG standalone and it worked.

Keep this in mind

Wilsker replied:
Did you check if the diagrams changed when setting the light quark masses to zero? Do you still get the diagrams with the Higgs mediator?

menglu21 replied:
i don't think there is sample even considering the Higgs-to-usd coupling, if you want to consider H-c/b coupling, you may need to switch to 3f or 4f scheme.

Wilsker replied:
The gridpack from these cards generates diagrams where for example, the Higgs couples with the initial state u quark. My question is whether these diagrams still exist when the light quark mass and Yukawa is set to zero?

menglu21 replied:
no such diagrams after setting light quark mass to zero

Wilsker replied:
This is problematic as it removes most of the signal diagrams (i.e those with a t-H coupling vertex)

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_proc_card.dat:
```diff
@@ -0,0 +1,42 @@
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+generate p b > j w- z t QCD=0 QED=99 QNP=2
+add process p b~ > j w+ z t~ QCD=0 QED=99 QNP=2
```
Hi, more importantly - this will give you asymmetric events.
You need to add the counter part "b p"

Wilsker replied:
1. Correct
2. I guess QNP is some old nomenclature? I've seen it used in various theorists cards. Is it problematic?

menglu21 replied:
i didn't notice that, i tested it using MG331, it may be correct in older version. it's fine if you already tested it.

Wilsker replied:
> Hi, more importantly - this will give you asymmetric events. You need to add the counter part "b p"

What do you mean by asymmetric? I think these describe the same contribution to the amplitude and considering both would be double counting. To check this, a while back I'd checked for changes in the diagrams generated for e.g. the " u b > u w- z t " process when generating " p p > j w- z t ". Although both the "u b > " and "b u > " subprocesses are there, the number of diagrams are the same, so I think this means Madgraph indeed considers them degenrate.

menglu21 replied:
hi, did you check the xs, if you do such test, define p = u u~, and generate three processes:
1. p u > e+ e-, xs=179pb
2. u p > e+ e-, xs=179pb
3. p p > e+ e-, xs=359pb

Wilsker replied:
You're right:
pb>tWZj
Cross-section :   0.003717 +- 3.288e-05 pb
pb>tWZj + bp>tWZj :
Cross-section :   0.007483 +- 4.984e-05 pb

I will add this to the card, thank you!

Wilsker replied:
The issue with adding these to processes is that the madspin step then takes a lot of memory. I tested running such a card in MG. I first asked Madspin to do leptonic decays only which worked fine. I then tried running for all Madspin decays while monitoring the memory. The program incrementally increased to 11G Virt 10.3G Res (last my memory monitoring script checked) before returning "Exiting due to channel error." Do you have any suggestions to reduce the memory required in the Madspin step? Otherwise, I suppose one solution is to run seperate cards for the top and anti-top diagrams like for single top processes.

agrohsje commented:
restriction card is in the model. please try it out with the central version and let me know in case of problems. thanks!

Wilsker commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_run_card.dat:
```diff
@@ -0,0 +1,285 @@
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+  lhapdf	= pdlabel ! PDF set
+  325300	= lhaid ! if pdlabel=lhapdf, this is the lhapdf number
```
Because the b quark is not in the multiparticle definition, the default will be 4FS in this case. Given the process we are trying to generate is always generated with a b quark in the initial state, a 4FS PDF will give a cross-section of 0. Explicitly defining the PDF like this was a work around for this problem.

menglu21 replied:
ok, i see. it makes sense.

menglu21 commented on bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/tWZj_NLO_5f/tWZj_NLO_5f_run_card.dat:
```diff
@@ -0,0 +1,285 @@
+#*********************************************************************
+  True	= use_syst ! Enable systematics studies
+#
+  systematics	= systematics_program ! none, systematics [python], SysCalc [depreceted, C++]
+['--mur=0.5,1,2', '--muf=0.5,1,2', '--pdf=errorset'] = systematics_arguments ! see: https://cp3.irmp.ucl.ac.be/projects/madgraph/wiki/Systematics#Systematicspythonmodule
+# Syscalc is deprecated but to see the associate options type'update syscalc'#*********************************************************************
+#  Additional hidden parameters
+#*********************************************************************
+  ['--mur=0.5,1,2', '--muf=0.5,1,2', '--pdf=errorset']	= systematics_arguments # Choose the argment to pass to the systematics command. like --mur=0.25,1,4. Look at the help of the systematics function for more details.
```
sorry, i didn't select the correct lines, from line279 to 285, some previous experience shows they may spoil the parser

agrohsje commented:
One comment. We need to distinguish between yukawa and mass. We can get vertices via the yukawa (you can check the ufo, didn't verify yet myself), so setting the mass to 0 doesn't mean 0 coupling.

Wilsker commented:
> One comment. We need to distinguish between yukawa and mass. We can get vertices via the yukawa (you can check the ufo, didn't verify yet myself), so setting the mass to 0 doesn't mean 0 coupling.

Right, but for a Yukawa coupling between the Higgs and fermion fields, I dont think it makes sense to set the mass to zero and not the Yukawa (which is proportional to the mass of the fermion)

agrohsje commented:
The point is to treat the particle in a consistent way between parton shower and matrix element. It makes perfectly sense from a MC perspective to set the mass to 0 and keep the yukawa. It is really just the treatment in the simulation.

menglu21 commented:
btw, this process should be LO, right? the name of cards are "NLO", please correct the name.

Wilsker commented:
Hi all, I've updated the cards etc. I've adressed all comments and also added some memory reducing commands as I was getting some angry messages from the lxplus team. Please check and make sure they are ok.

menglu21 commented:
Hi @Wilsker you want to keep all the quark yukawa, right? the model you imported already set all mass and yukawa of udscb zero. so you need to import model Yt_UFO rather than Yt_UFO-nobmass.

Wilsker commented:
Hi @menglu21, we wanted to set the masses to zero but leave the Yukawa's untouched, so I still want to call the no-b-mass restriction card, I just don't want the Yukawas in the model set to zero. I thought @sihyunjeon was changing the model to set masses for b and lighter quarks to zero anf set the Yukawas for all, including the b, to the SM value?

menglu21 commented:
actually no, if you check the no-b-mass restriction card, you'll see mass and yukawa coupling of udscb are zero. The name "no-b-mass" is just gen convention that make mass and Yukawa consistent. In your case, you want all the Yukawa values to be SM ones, you can directly use the default model, i.e., without the restriction, but with setting mb=0 manually in the customized card.

sihyunjeon commented:
ah sorry for the mess. I did not know this was pinged to me 2 weeks ago. I have to fix the UFO for @Wilsker. I'll do it today right after lunch time

Wilsker commented:
> actually no, if you check the no-b-mass restriction card, you'll see mass and yukawa coupling of udscb are zero. The name "no-b-mass" is just gen convention that make mass and Yukawa consistent. In your case, you want all the Yukawa values to be SM ones, you can directly use the default model, i.e., without the restriction, but with setting mb=0 manually in the customized card.

Ok, this isn't the case for the restriction card I wrote so I think it was changed in the model directory. In anycase, I thought the recommended way to do this was to set masses = 0 and Yukawas to SM in a restriction card in the model to ensure the correct PDFs are used?

Wilsker commented:
> ah sorry for the mess. I did not know this was pinged to me 2 weeks ago. I have to fix the UFO for @Wilsker. I'll do it today right after lunch time

Hi @sihyunjeon , has this been sorted out yet? I believe everything else is now good to merge right @menglu21?

sihyunjeon commented:
Sorry I forgot to ping you here. merging
